behavior/modeling/model.py

```diff
@@ -23,82 +21,126 @@
 from behavior.modeling import METHODS
 
 
-def get_model(method: str, config: dict[str, Any]) -> Any:
+def get_model(method, config):
+    """Initialize and return the underlying Behavior Model variant with the provided configuration parameters.
+
+    Parameters
+    ----------
+    method : str
+        Regression model variant.
+    config : dict[str, Any]
+        Configuration parameters for the model.
+
+    Returns
+    -------
+    model : Any
+       A regression model.
+
+    Raises
+    ------
+    ValueError
+        If the requested method is not supported.
+    """
     if method not in METHODS:
```

Comment:
Technically, as a code style thing, a check like this goes against the Python "easier to ask forgiveness than permission" philosophy. It also creates annoying error callstacks in larger projects, if people start to lose track of why the check is there and if the check is nonlocal to the constraint, they tend to get out of sync.

Not a big deal though.